### PR TITLE
[DX] Cleanup ComplexNodeRemover

### PR DESCRIPTION
**Pull request by TomasVotruba**
Follow up to https://github.com/rectorphp/rector-src/pull/2229

### CODE REVIEW

File: rules-tests/DeadCode/Rector/Property/RemoveUnusedPrivatePropertyRector/Fixture/constructor_with_middle_params.php.inc [hunk 1]
```diff
@@ -27,9 +27,9 @@ class DoNotRemoveParameterOnConstructorOnlyInMiddleParams
 
 namespace Rector\Tests\DeadCode\Rector\Property\RemoveUnusedPrivatePropertyRector\Fixture;
 
-class DoNotRemoveParameterOnConstructorOnlyInMiddleParams
+final class ConstructorWithMiddleParams
 {
-    public function __construct(int $contentFinder, \stdClass $stdClass)
+    public function __construct(\stdClass $stdClass)
```
Comment (samsonasik) [inline rules-tests/DeadCode/Rector/Property/RemoveUnusedPrivatePropertyRector/Fixture/constructor_with_middle_params.php.inc:32]: the `$contentFinder` paraam need to be kept as it is in the first param, which cause a BC break, that's why there is test for `DoNotRemoveParameterInMiddle` test fixture.

Comment (TomasVotruba) [reply rules-tests/DeadCode/Rector/Property/RemoveUnusedPrivatePropertyRector/Fixture/constructor_with_middle_params.php.inc:32]: I see. Could you extend this PR to keep this behavior? There was a lot of unneded checks that could be removed.

Comment (samsonasik) [reply rules-tests/DeadCode/Rector/Property/RemoveUnusedPrivatePropertyRector/Fixture/constructor_with_middle_params.php.inc:32]: I will try, do you want me to create separate PR after it got merged?

File: rules/Removing/NodeManipulator/ComplexNodeRemover.php [hunk 2]
```diff
@@ -310,4 +185,31 @@ private function resolvePropertyFetchFromDimFetch(Expr $expr): array
 
         return $propertyFetches;
     }
+
+    private function clearParamFromConstructor(ClassMethod $classMethod, Variable $assignedVariable): void
+    {
+        // is variable used somewhere else? skip it
+        $variables = $this->betterNodeFinder->findInstanceOf($classMethod, Variable::class);
```
Comment (samsonasik) [inline rules/Removing/NodeManipulator/ComplexNodeRemover.php:192]: there is `findInstancesOfInFunctionLikeScoped` for it:

```suggestion
        $variables = $this->betterNodeFinder->findInstancesOfInFunctionLikeScoped($classMethod, [Variable::class]);
```

so no need to get from inner function stmts.

File: rules-tests/DeadCode/Rector/Property/RemoveUnusedPrivatePropertyRector/Fixture/constructor_with_middle_params.php.inc (hunk 1 above)
Comment (TomasVotruba) [reply rules-tests/DeadCode/Rector/Property/RemoveUnusedPrivatePropertyRector/Fixture/constructor_with_middle_params.php.inc:32]: That would be the best 👍 
I fell asleep yesterday waiting for CI, so I merge now.

Comment (samsonasik) [reply rules-tests/DeadCode/Rector/Property/RemoveUnusedPrivatePropertyRector/Fixture/constructor_with_middle_params.php.inc:32]: https://github.com/rectorphp/rector-src/pull/2231

**Pull request merged**